src/config.ts

```diff
@@ -55,3 +44,51 @@ export async function getConfig(flags: Flags) {
   config = Object.assign({}, config, strippedFlags);
   return config;
 }
+
+const validConfigExtensions = ['.js', '.mjs', '.cjs', '.json'];
+type ConfigExtensions = typeof validConfigExtensions[number];
+
+async function parseConfigFile(configPath: string): Promise<Flags> {
+  const typeOfConfig = getTypeOfConfig(configPath);
+
+  switch (typeOfConfig) {
+    case '.json':
+      return readJsonConfigFile(configPath);
+    case '.js':
+    case '.mjs':
+    case '.cjs':
+      return importConfigFile(configPath);
+  }
+
+  throw new Error(`Config file ${configPath} is invalid`);
+}
+
+function getTypeOfConfig(configPath: string): ConfigExtensions {
+  // Returning json in case file doesn't have an extension for backward compatibility
+  const configExtension = path.extname(configPath) || '.json';
+
+  if (validConfigExtensions.includes(configExtension)) {
+    return configExtension as ConfigExtensions;
+  }
+
+  throw new Error(
+    `Config file should be either of extensions ${validConfigExtensions.join(
+      ','
+    )}`
+  );
+}
+
+async function importConfigFile(configPath: string): Promise<Flags> {
+  const config = (await import(path.resolve(process.cwd(), configPath)))
+    .default;
+
+  return config;
+}
+
+async function readJsonConfigFile(configPath: string): Promise<Flags> {
+  const configFileContents: string = await fs.readFile(configPath, {
```

Comment:
No need for the `: string` here, it's fine implicit

Reply:
sure, makes sense, removed extra `: string`

Reply:
This is done @JustinBeckwith